codegen/src/serialization.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::BTreeMap;
```

Comment:
Would you like me to add a comment to the top of the file about this, just so it's clear in the future?